cmake_modules/PaRSECConfig.cmake.in

```diff
@@ -20,7 +20,9 @@ find_package(Threads)
 
 if(@PARSEC_HAVE_HWLOC@)
   set_and_check(HWLOC_INCLUDE_DIR "@HWLOC_INCLUDE_DIR@")
+  set(HWLOC_INCLUDE_DIR ${HWLOC_INCLUDE_DIR} CACHE PATH "Set by PaRSECConfig.cmake" FORCE)
   set_and_check(HWLOC_LIBRARY "@HWLOC_LIBRARY@")
+  set(HWLOC_LIBRARY ${HWLOC_LIBRARY} CACHE PATH "Set by PaRSECConfig.cmake" FORCE)
   find_package(HWLOC REQUIRED)
```

Comment:
My understanding was that we are not supposed to force these variable in the cache, but instead they will be put in the cache by find_package.

Reply:
Reading again on the find_package documentation, maybe the only thing we need is to define HWLOC_DIR (not HWLOC_ROOT) and allow find_package to populate the rest.

Reply:
HWLOC_DIR is used to find hwloc-config.cmake (that's it's only use), so that's not a good option. 

The reason why we set these in cache by force is because we have already done the findHWLOC in the past, and we want to force the same library (hence forcing the outcome of the detection the second time around).

Reply:
I tried to understand this a little, and I think the root cause is different than what stated here. It appear that the VARIABLE used in find_path is overwritten if not in the cache upon entry in find_path. However, the solution proposed here, aka forcing the VARIABLE to be in the cache, results in no test done, the find_path is considered successful simply because the cached variable exists.

Reply:
Let's assume there is no reason for HWLOC_INCLUDE_DIR and HWLOC_LIBRARY to be defined in the cache if they were not found by a prior search for HWLOC.